### PR TITLE
Increase max tag value length to 1024 and make it configurable

### PR DESCRIPTION
**Pull request by yurishkuro**
Resolves https://github.com/jaegertracing/jaeger/issues/508

### CODE REVIEW

File: jaeger_client/constants.py [hunk 1]
```diff
@@ -65,3 +65,6 @@
 # of traces per second.
 # noinspection SpellCheckingInspection
 SAMPLER_TYPE_LOWER_BOUND = 'lowerbound'
+
+# max length for tag values. Longer values will be truncated.
+MAX_TAG_VALUE_LENGTH = 1024
```
Comment (vprithvi) [inline jaeger_client/constants.py]: Why 1024? 
What is the purpose of curtailing the length? 
Is it better to have `MAX_TAG_VALUE_BYTES` instead? (Because unicode chars can span multiple bytes?)

File: tests/test_reporter.py [hunk 2]
```diff
@@ -225,7 +225,7 @@ def test_submit_queue_full_batch_size_1(self):
 
     @gen_test
     def test_submit_batch_size_2(self):
-        reporter, sender = self._new_reporter(batch_size=2, flush=0.005)
+        reporter, sender = self._new_reporter(batch_size=2, flush=0.01)
```
Comment (vprithvi) [inline tests/test_reporter.py]: Why is this change required in this diff?

File: tests/test_span.py [hunk 3]
```diff
@@ -232,3 +232,12 @@ def test_span_component(tracer):
 
     span.set_tag(ext_tags.COMPONENT, 'crypt')
     assert span.component == 'crypt'
+
+
+def test_span_tag_value_max_length(tracer):
+    tracer.max_tag_value_length = 42
+    span = tracer.start_span(operation_name='x')
+    span.set_tag('x', 'x' * 50)
```
Comment (vprithvi) [inline tests/test_span.py]: Could you add a test with unicode as well?

File: jaeger_client/constants.py (hunk 1 above)
Comment (yurishkuro) [reply jaeger_client/constants.py]: python client does not have logic to calculate the size of the UDP packet, instead it just limits the number of spans per packet. Having unbounded tag/log values increases the risk of exceeding the packet size. We've seen huge SQL statements in production.

File: tests/test_reporter.py (hunk 2 above)
Comment (yurishkuro) [reply tests/test_reporter.py]: the test was unstable when it relied on the difference between 1ms and 5ms

File: tests/test_span.py (hunk 3 above)
Comment (yurishkuro) [reply tests/test_span.py]: String truncation works fine with unicode. I don't see a point of testing a feature of the language.

```
>>> x=u'key2-caf\xe9-blah'
>>> print(x)
key2-café-blah
>>> print(x[:10])
key2-café-
>>> print(x[:9])
key2-café
>>> print(x[:8])
key2-caf
```

File: jaeger_client/constants.py (hunk 1 above)
Comment (vprithvi) [reply jaeger_client/constants.py]: If we want to figure out the size in bytes, I think it is more useful to have `MAX_TAG_VALUE_BYTES` instead of length, because the length would be shorter than the number of bytes when using some unicode characters.

Comment (yurishkuro) [reply jaeger_client/constants.py]: why does it matter? It's an approximation anyway.

Comment (vprithvi) [reply jaeger_client/constants.py]: I see, I didn't realize that it was an approximation.
